packages/driver/src/cypress/command_queue.ts

```diff
@@ -247,7 +249,7 @@ export class CommandQueue extends Queue<$Command> {
       // we're finished with the current command so set it back to null
       this.state('current', null)
 
-      this.state('subject', subject)
```

Comment:
This PR removes `cy.state('subject')`, right? I think we should be careful here - it's not a publicly-documented API, but it seems like it's used in a couple of Cypress plugins, testing-library/cypress-testing-library and Lakitna/cypress-commands (tests only):

https://github.com/search?q=cy.state%28%27subject%27%29&type=code

What do you think about still setting `cy.state('subject')` just for backwards-compatibility, and opening a second PR for 11.0.0 that will remove it entirely, so the plugins have some time to update? `cypress-testing-library` gets 3MM npm downloads a month, so I'm worried about breaking those users.

Reply:
This PR changes the formatting of it, to be an object rather than a single entry, in preparation for changing it even further down the line (into an even more complex data structure). You're right though, if testing-library is relying on it, then this is a breaking change and I absolutely cannot make it as part of a `chore`.

Hm. This is really rough though. The call `cy.state('subject')` is exactly what I'm trying to get rid of - it's the source of detached DOM errors, distilled to its essence. Let me think on this a bit and figure out what to do here.

Reply:
Ok, so thinking about this further, I think we can in fact leave `cy.state('subject')` 'mostly working', and add a deprecation warning to it. The basic problem is that the *it does not map cleanly to the current subject*, and never has. I guarantee you every third-party project using `cy.state('subject')` has subtle bugs around the use of `cy.within` and `cy.as`.

That's what I mean when I say "mostly working" - I can add a shim to keep it working as-is (only a little bit broken).

Reply:
So current plan:
1. Add a shim on `cy.state('subject')` in this PR so it keeps working as a `chore`. Merge this PR.
2. In next PR, introduce selectors / function chains as subject, as a `feat`. Keep the shim working.
3. Open a PR with `cypress-testing-library` to move them to the new functionality.
4. Remove shim in breaking 11.x release eventually.

Reply:
Nice catch here. We should probably make a PR against testing-library once this lands, to save them the time/effort, and while this is fresh in our minds.

Reply:
Does PR likely needs to be flagged for release so a note can be made in the changelog that this usage has been deprecated if we are porting this functionality?

Reply:
Good call. I've updated the PR description to note the change and the rational behind it.